test/helpers/integration.rb

```diff
@@ -460,27 +460,36 @@ def hot_restart_does_not_drop_connections(num_threads: 1, total_requests: 500)
 
     run = true
 
+    thread_error = nil
+
     restart_thread = Thread.new do
       sleep 0.2  # let some connections in before 1st restart
-      while run
-        if Puma.windows?
-          cli_pumactl 'restart'
-        else
-          Process.kill :USR2, @pid
-        end
-        sleep 0.5
-        # If 'wait_for_server_to_boot' times out, error in thread shuts down CI
-        begin
-          wait_for_server_to_boot timeout: 5
-        rescue Minitest::Assertion # Timeout
-          run = false
+      begin
+        while run
+          if Puma.windows?
+            cli_pumactl 'restart'
+          else
+            Process.kill :USR2, @pid
+          end
+          sleep 0.5
+          # If 'wait_for_server_to_boot' times out, error in thread shuts down CI
+          begin
+            wait_for_server_to_boot timeout: 5
+          rescue Minitest::Assertion # Timeout
+            run = false
+          end
+          restart_count += 1
+          sleep(Puma.windows? ? 2.0 : 0.5)
         end
-        restart_count += 1
-        sleep(Puma.windows? ? 2.0 : 0.5)
+      rescue StandardError => e
+        thread_error = e
       end
     end
 
+    fail "Error in restart_thread #{thread_error.class}" if thread_error
```

Comment:
This will provide a bit more info, could be useful?

```suggestion
    fail "Error in restart_thread #{thread_error.inspect}" if thread_error
```

Reply:
@dentarg Thanks & ok.

I changed the PR somewhat after consuming a proper amount of coffee.  I'm running it a few times in my fork with only `test_integration_cluster.rb` and `test_integration_single.rb`, which are the two files that use `hot_restart_does_not_drop_connections`.

Reply:
Changed line to:
```ruby
flunk "Error in restart_thread\n#{thread_error.inspect}" if thread_error
```